app/models.py

```diff
@@ -60,3 +125,46 @@ def decorated(*args, **kwargs):
         return func(current_user, *args, **kwargs)
 
     return decorated
+
+class User(object):
+    """ Handle user """
+
+    def __init__(self):
+        self.db = CONN.db_return()
+
+    def register_user(self, username, user_email, user_password):
+        """ registers a  new user"""
+        checker = self.db.cursor()
+        checker.execute("SELECT username, email FROM users")
+        for user in checker.fetchall():
+            # if username == user[0]:
+            #     return {"status": "fail", "message" : "user exists"}, 409
+            if user_email == user[1]:
+                return {"status": "fail", "message" : "user exists"}, 409
+
+        cur = self.db.cursor()
+        query = "INSERT INTO users (email, password, username) VALUES (%s, %s, %s)"
+        hashed_password = generate_password_hash(user_password, method='sha256')
+        data = (user_email, hashed_password, username)
+        cur.execute(query, data)
+        self.db.commit()
+        response = {"status": "success", 
+            "Registered": {"Email":str(user_email), 
+                            "Username":str(username)}}
+        return response, 201
+    
+    def login_user(self, user_email, user_password):
+        checker = self.db.cursor()
+        checker.execute("SELECT * FROM users")
```

Comment:
You could just get the specific user from the database, Its just added latency on the database when you get all users just so you could filter for one user. It is a very expensive operation on the database side. Use the WHERE clause to get just the user you need. https://www.w3schools.com/sql/sql_where.asp

Reply:
That is much easier thanks you so much :100: